### PR TITLE
add bed12 recipe

### PR DESCRIPTION
**Pull request by mirpedrol**
Add new recipe for bed12 and new asset class for bed file format as mentioned in [#271](https://github.com/refgenie/refgenie/pull/271).
Required on [refgenie/aws_igenomes #1](https://github.com/refgenie/aws_igenomes/pull/1)

### CODE REVIEW

**Comment by ewels** [comment]
@nsheff - does this look good? I think this is the only thing remaining for us to get the RefGenie iGenomes on parity with AWS-iGenomes now.

**Comment by nsheff** [comment]
Ok great -- I will finalize this very soon

**Comment by ewels** [comment]
Hi @nsheff - anything we can do to move this PR along? 🙏🏻

File: recipes/bed12_asset_recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: bed12
+version: 0.0.1
+output_asset_class: bed
+description: Genomic feature annotations in BED12 format
+inputs:
+  files:
+    bed:
+      description: Genomic feature annotations in Browser Extensible Data (BED12) format
```
Comment (nsheff) [inline recipes/bed12_asset_recipe.yaml:8]: The input file is actually expected to be in gzipped bed format, right?

**Comment by nsheff** [comment]
I updated the key to `bedgz` to reflect that you're actually wanting to store a gzipped bed file

**Comment by ewels** [comment]
Nope, we've been using raw bed files - not compressed - until now (see path [here](https://github.com/nf-core/rnaseq/blob/e049f51f0214b2aef7624b9dd496a404a7c34d14/conf/igenomes.config#L21) for example).

But you're right, we should probably be using gzipped files. Means updating all of the code using these references but shouldn't be too difficult as long as we remember.

**Comment by nsheff** [comment]
Ok -- I ask because the recipe was naming the file `.bed.gz`, so this wouldn't have worked with raw .bed files... well, they would have been bed files named `blah.bed.gz`...

**Comment by ewels** [comment]
Oh I see 😅 Ok I'm sure it's fine, as long as @mirpedrol is on top of things 👍🏻 

x-ref change here: https://github.com/refgenie/recipes/commit/0bf4b6f0b39fe3313b140a3715da31dc92f38812

**Comment by mirpedrol** [comment]
Thanks for the merge! 😄

**Pull request merged**